libindy/src/services/pool/request_handler.rs

```diff
@@ -945,6 +928,16 @@ pub mod tests {
             assert_match!(RequestState::Finish(_), request_handler.request_wrapper.unwrap().state);
         }
 
+        #[test]
+        fn request_handler_process_reply_event_from_consensus_state_works_for_consensus_reached_with_mixed_msgs() {
```

Comment:
I have some concerns about this test.
First we create a request handler with 1 node and `f = 0`. Then we send to it 3 responses -- it just doesn't make sense.
What I suggest to do is to write a test that would fail before and pass right now.
Example:
* Create a request handler with 8 nodes and `f = 2`
* Start with sending a `CustomConsensusEvent`
* Send one `Reply` and two `Reject`messages with different data in each `Reject`
* Check that state of request handler is still not finished
* Send one more `Reject` with data that equals to one of the previous `Reject` messages
* Check that state of request handler changed to `Finish`

I think that this one will be enough to verify the changes made for consensus case and reject. The same test should implemented for renack and both of them for single case.